pallets/creditcoin/src/types/platform.rs

```diff
@@ -20,38 +36,100 @@ impl From<u64> for EvmChainId {
 }
 
 impl EvmChainId {
-	pub fn new(value: u64) -> Self {
+	pub const fn new(value: u64) -> Self {
 		EvmChainId(value)
 	}
+	pub fn as_u64(self) -> u64 {
+		self.0
+	}
+
+	pub const ETHEREUM: EvmChainId = EvmChainId::new(1);
+	pub const RINKEBY: EvmChainId = EvmChainId::new(4);
+	pub const LUNIVERSE_TESTNET: EvmChainId = EvmChainId::new(949790);
+	pub const LUNIVERSE: EvmChainId = EvmChainId::new(59496427);
 }
 
 #[derive(
-	Clone, RuntimeDebug, PartialEq, Eq, PartialOrd, Encode, Decode, TypeInfo, MaxEncodedLen,
+	Clone, RuntimeDebug, PartialEq, Eq, PartialOrd, Ord, Encode, Decode, TypeInfo, MaxEncodedLen,
 )]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "std", serde(rename_all = "camelCase"))]
 pub struct EvmInfo {
 	pub chain_id: EvmChainId,
 }
 
+impl EvmInfo {
+	pub const ETHEREUM: EvmInfo = EvmInfo { chain_id: EvmChainId::ETHEREUM };
+	pub const RINKEBY: EvmInfo = EvmInfo { chain_id: EvmChainId::RINKEBY };
+	pub const LUNIVERSE_TESTNET: EvmInfo = EvmInfo { chain_id: EvmChainId::LUNIVERSE_TESTNET };
+	pub const LUNIVERSE: EvmInfo = EvmInfo { chain_id: EvmChainId::LUNIVERSE };
+}
+
 #[derive(
 	Clone, RuntimeDebug, PartialEq, Eq, PartialOrd, Encode, Decode, TypeInfo, MaxEncodedLen,
 )]
-pub enum NewBlockchain {
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "std", serde(rename_all = "camelCase"))]
+pub enum Blockchain {
 	Evm(EvmInfo),
 }
 
+impl Blockchain {
+	pub const fn evm(chain_id: EvmChainId) -> Blockchain {
+		Blockchain::Evm(EvmInfo { chain_id })
+	}
+	// Chain IDs from chainlist.org and Gluwa's internal luniverse documentation
+	pub const ETHEREUM: Blockchain = Blockchain::evm(EvmChainId::ETHEREUM);
+	pub const RINKEBY: Blockchain = Blockchain::evm(EvmChainId::RINKEBY);
+	pub const LUNIVERSE_TESTNET: Blockchain = Blockchain::evm(EvmChainId::LUNIVERSE_TESTNET);
+	pub const LUNIVERSE: Blockchain = Blockchain::evm(EvmChainId::LUNIVERSE);
+
+	pub fn as_bytes(&self) -> Cow<'_, [u8]> {
+		match *self {
+			Blockchain::ETHEREUM => Cow::Borrowed(&*b"ethereum"),
+			Blockchain::RINKEBY => Cow::Borrowed(&*b"rinkeby"),
+			Blockchain::LUNIVERSE_TESTNET | Blockchain::LUNIVERSE => Cow::Borrowed(b"luniverse"),
+			Blockchain::Evm(EvmInfo { chain_id }) => {
+				let s = chain_id.as_u64().to_string();
+				let mut buf = b"evm-".to_vec();
+				buf.extend(s.as_bytes());
+				Cow::Owned(buf)
+			},
+		}
+	}
+
+	pub fn supports(&self, kind: &LegacyTransferKind) -> bool {
```

Comment:
No, because the supported transfer kinds for a given chain are encoded directly in the `Currency` type.
```rust
pub enum EvmTransferKind {
	Erc20,
	Ethless,
}
pub type EvmSupportedTransferKinds =
	BoundedVec<EvmTransferKind, ConstU32<{ EvmTransferKind::COUNT as u32 }>>;
pub enum EvmCurrencyType {
	SmartContract(ExternalAddress, EvmSupportedTransferKinds),
}
pub enum Currency {
	Evm(EvmCurrencyType, EvmInfo),
}

```
So it's impossible to construct a currency with a transfer kind that isn't supported.